config/config.go

```diff
@@ -114,13 +115,19 @@ func (cfg *Config) NewNode(ctx context.Context) (host.Host, error) {
 		return nil, err
 	}
 
+	// create an event bus
+	eb := eventbus.NewBus()
+
 	// TODO: Make the swarm implementation configurable.
-	swrm := swarm.NewSwarm(ctx, pid, cfg.Peerstore, cfg.Reporter)
+	swrm, err := swarm.NewSwarm(ctx, pid, cfg.Peerstore, cfg.Reporter, eb)
+	if err != nil {
+		return nil, err
+	}
 	if cfg.Filters != nil {
 		swrm.Filters = cfg.Filters
 	}
 
-	h, err := bhost.NewHost(ctx, swrm, &bhost.HostOpts{
+	h, err := bhost.NewHost(ctx, swrm, eb, &bhost.HostOpts{
```

Comment:
Can we make this a host option (and let the host construct it if it's nil?). Slightly less breaking that way.

Reply:
@Stebalien  _For my own understanding_

How do breaking changes like these affect us/our users considering we aren't really breaking an interface ?

Reply:
Technically, users might be manually constructing their hosts (without using the constructor). That used to the the correct way to do this until we introduced the constructor. I don't _think_ anyone is, but I haven't checked thoroughly.

But really, either way is probably fine in this case.